app/graphql/mutations/sign_in_user.rb

```diff
@@ -13,8 +13,22 @@ class SignInUser < BaseMutation
     def resolve(username:, password:)
       return unless username && password
 
+      bruteforce_protection = Foreman::BruteforceProtection.new(
+        request_ip: context[:request_ip]
+      )
+
+      if bruteforce_protection.bruteforce_attempt?
+        bruteforce_protection.log_bruteforce
+        raise GraphQL::ExecutionError, _('Too many tries, please try again in a few minutes.')
```

Comment:
should this be translated or just extracted? If it's an exception user will be facing, maybe the exception class should inherit from Foreman::Exception (I didn't check if it isn't the case already)

Reply:
I think we're good here.

> If a field's resolve function returns a ExecutionError, the error will be inserted into the response's `“errors”` key and the field will resolve to `nil`.(https://www.rubydoc.info/github/rmosolgo/graphql-ruby/GraphQL/ExecutionError)